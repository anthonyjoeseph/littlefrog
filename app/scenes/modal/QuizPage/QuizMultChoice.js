// app/QuizMultChoice.js

import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View,
  TextInput,
  TouchableWithoutFeedback
} from 'react-native';

import Button from 'react-native-button'
import { RadioButtons } from 'react-native-radio-buttons';

const options = [
  "Option A",
  "Option B",
  "Option C",
  "Option D"
];

class QuizMultChoice extends Component {
  constructor(props) {
    super(props)

    this.state = {
      selectedOptionIndex: -1
    }

    this.setSelectedOption = this.setSelectedOption.bind(this);
    this.renderOption = this.renderOption.bind(this);
    this.renderContainer = this.renderContainer.bind(this);
    this.answer = this.answer.bind(this);
  }

  setSelectedOption(selectedOption){
    var selectedOptionIndex = this.props.options.indexOf(selectedOption);
    this.setState({
      selectedOption,
      selectedOptionIndex: selectedOptionIndex
    });
  }

  renderOption(option, selected, onSelect, index){
    const style = selected ? { fontWeight: 'bold' } : {};

    return (
      <TouchableWithoutFeedback onPress={onSelect} key={index}>
        <View>
          <Text style={style}>{option}</Text>
        </View>
      </TouchableWithoutFeedback>
    );
  }

  renderContainer(optionNodes){
    return <View>{optionNodes}</View>;
  }

  answer(){
    if(this.state.selectedOptionIndex == this.props.correctIndex){
      this.props.onCorrect();
    }else{
      this.props.onWrong();
    }
  }

  render() {
    return (
      <View>
        <Text style={{fontSize:25}}>{this.props.question}</Text>
        <RadioButtons
          options={ this.props.options }
          onSelection={ this.setSelectedOption }
          selectedOption={this.state.selectedOption }
          renderOption={ this.renderOption }
          renderContainer={ this.renderContainer }
        />
        <Text style={{color:'rgb(150, 150, 150)'}}>Selected option: {this.state.selectedOption || 'none'}</Text>
        <Button onPress={this.answer}>ANSWER</Button>
      </View>
    )
  }
}

export default QuizMultChoice;
