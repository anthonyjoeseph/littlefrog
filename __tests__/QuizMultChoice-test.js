// __tests__/DismissableFullscreenPopup-test.js
import 'react-native';
import React from 'react';
import QuizMultChoice from '../app/scenes/modal/QuizPage/QuizMultChoice';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <QuizMultChoice
      question="What is my favorite food?"
      options={[
        "Bananas",
        "Razors",
        "Fried Chicken",
        "Pencil Shavings"
      ]}
      correctIndex={2}
      onCorrect={ ( ) => { } }
      onWrong={ ( ) => { } }
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
