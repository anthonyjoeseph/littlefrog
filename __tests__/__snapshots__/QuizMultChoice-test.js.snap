exports[`test renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 25,
      }
    }>
    What is my favorite food?
  </Text>
  <View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={undefined}
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={Object {}}>
        Bananas
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={undefined}
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={Object {}}>
        Razors
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={undefined}
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={Object {}}>
        Fried Chicken
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={undefined}
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={Object {}}>
        Pencil Shavings
      </Text>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "rgb(150, 150, 150)",
      }
    }>
    Selected option: 
    none
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}>
    <View
      style={
        Array [
          Object {},
          Object {},
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0C42FD",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#841584",
            },
            Object {
              "color": "#cdcdcd",
            },
          ]
        }>
        ANSWER
      </Text>
    </View>
  </View>
</View>
`;
